components/nimbus/src/behavior.rs

```diff
@@ -0,0 +1,415 @@
+#![allow(dead_code)]
+
+use crate::error::{BehaviorError, NimbusError, Result};
+use chrono::{DateTime, Timelike, Utc};
+use std::collections::{HashMap, VecDeque};
+use std::hash::{Hash, Hasher};
+
+#[derive(Clone)]
+pub enum Interval {
```

Comment:
If you had something like:

```
impl fmt::Display for Interval {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        fmt::Debug::fmt(self, f)
    }
}
```

Then you could call `to_string()` to get a `String` representation of the enum variants and maybe you wouldn't need the `IntervalName` stuff below.